src/sw.ts

```diff
@@ -94,7 +95,32 @@ self.addEventListener('fetch', (event) => {
     // intercept and do our own stuff...
     event.respondWith(fetchHandler({ path: url.pathname, request }))
   } else if (isSubdomainRequest(event)) {
-    event.respondWith(fetchHandler({ path: url.pathname, request }))
+    const cacheKey = event.request.url
+    log(`helia-sw: cache key:`, cacheKey.toString())
+
+    event.respondWith((async () => {
+      const cache = await caches.open(IPFS_CACHE)
+      const cachedResponse = await cache.match(cacheKey)
+
+      if (cachedResponse) {
+        // If there is an entry in the cache for event.request,
+        // then response will be defined and we can just return it.
+        // Note that in this example, only font resources are cached.
+        log('helia-ws: cached response for %s found in cache: %o', cacheKey, cachedResponse)
+        return cachedResponse
+      }
+
+      // 👇 fetch because no cached response was found
+      const response = await fetchHandler({ path: url.pathname, request })
+
+      if(response.ok) {
+        // 👇 only cache successful responses
+        log('helia-ws: storing cache key %s in cache', cacheKey)
+        // Clone the response since streams can only be consumed once.
+        cache.put(cacheKey, response.clone())
```

Comment:
This works for immutable `/ipfs/cid`, but needs separate logic for mutable responses.

One way of implementing expiration on `/ipns/..` is to add `Expires` header to a cached `/ipns/..` response (everything other than immutable path or subdomain), and then when it is read back via `cache.match` check if expiration happenend. If it expired, fetch new one from network. We can switch it to stale-while-revalidate later by always returning cached result and doing update async.